components/common/Row.tsx

```diff
@@ -1,21 +1,17 @@
 import React from 'react';
-import type { StyleProp, ViewStyle } from 'react-native';
-import { StyleSheet, View as NativeView } from 'react-native';
+import { StyleSheet, View } from 'react-native';
 
 type RowProps = {
   children: React.ReactNode;
-  style?: StyleProp<ViewStyle>;
+  style?: object;
 };
 
-/**
- * @param style: StyleProp<TextStyle> - custom style
- * @param children: React.ReactNode - text
- */
-
-const Row = (props: RowProps & NativeView['props']) => {
-  const { style } = props;
-  const viewStyle = [styles.row, style];
-  return <NativeView style={viewStyle} {...props} />;
+const Row = ({ children, style }: RowProps) => {
+  return (
+    <View testID="row" style={[styles.row, style]}>
```

Comment:
Looking at the rendered JSX, there are a lot of `undefined` in the stylesheets because passing `style` props to the components is optional. For example:

```tsx
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
  testID="row"
>
  <View />
</View>
```

We should use `StyleSheet.compose` to compose the stylesheets when receiving custom styles. This will combine the styles and perform all overrides nicely. If the custom style prop is not provided, the default style in the component is returned without allocating an array. It will look like this:

```tsx
<View testID="row" style={StyleSheet.compose(styles.row, style)}>
  {children}
</View>
```

Which will render to:

```tsx
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
  testID="row"
>
  <View />
</View>
```

I suggest we use `StyleSheet.compose` in all components that accept custom styles (`Row`, `MainContainer`, `BottomSheetContainer`). This will also make the code more consistent.

Reply:
done